.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+# Name of the workflow
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request 
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  end-to-end:
+    strategy:
+      matrix:
+        test: [Project1Test, Project2Test]
+    runs-on: ubuntu-latest
+
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '13' # The JDK version to make available on the path.
+        java-package: jdk # (jre, jdk, or jdk+fx) - defaults to jdk
+        architecture: x64 # (x64 or x86) - defaults to x64
+    - name: Set up libraries
+      run: |
+        mkdir libraries
+        cd libraries
+        curl -O https://repo1.maven.org/maven2/org/apache/opennlp/opennlp-tools/1.9.2/opennlp-tools-1.9.2.jar
+        curl -O https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-core/2.13.0/log4j-core-2.13.0.jar
+        curl -O https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-api/2.13.0/log4j-api-2.13.0.jar
+        curl -O https://repo1.maven.org/maven2/org/junit/platform/junit-platform-console-standalone/1.6.0-RC1/junit-platform-console-standalone-1.6.0-RC1.jar
+        curl -O https://repo1.maven.org/maven2/org/eclipse/jetty/aggregate/jetty-all/9.4.26.v20200117/jetty-all-9.4.26.v20200117-uber.jar
+        curl -O https://repo1.maven.org/maven2/org/apache/commons/commons-text/1.8/commons-text-1.8.jar
+        curl -O https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.9/commons-lang3-3.9.jar
+        curl https://repo1.maven.org/maven2/org/antlr/antlr4/4.7.2/antlr4-4.7.2-complete.jar > antlr-4.7.2-complete.jar
```

Comment:
On first review, we decided that I should redownload the dependencies from a canonical source every time that we used them, rather than have a "libraries" repository.

However, this presented one potential problem. The "ST-4.2.jar" file found in the "libraries" directory used by the project script compiles fine with Java 13, but the "ST-4.2.jar" file found on the StringTemplate website doesn't seem to work with later versions of Java and gives me an error when compiling the same way.

So this commit doesn't include that jar file. Will we need that?

Reply:
No, StringTemplate does not need to be included.

The unknown is whether we will need JDBC. With the delay, I'm not sure we will get to that point this semester.